lib/tokenlist.h

```diff
@@ -68,7 +68,8 @@ class CPPCHECKLIB TokenList {
     /** @return true if the code is C++ */
     bool isCPP() const;
 
-    void setLang(Standards::Language lang);
+    // TODO: get rid of this
+    void setLang(Standards::Language lang, bool force = false);
```

Comment:
This function is just a hack. The only other user is the Clang import because our unit tests use data in an incomplete format.

Also all this logic and asserts will (hopefully) go away when we perform the file identification early on for all input instead of on-demand when it is tokenized.